kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/PartitionLog.java

```diff
@@ -526,6 +526,11 @@ public CompletableFuture<ReadRecordsResult> readRecords(final FetchRequest.Parti
                 requestStats.getPrepareMetadataStats().registerSuccessfulEvent(
                         MathUtils.elapsedNanos(startPrepareMetadataNanos), TimeUnit.NANOSECONDS);
                 long adjustedMaxBytes = Math.min(partitionData.maxBytes, limitBytes.get());
+                if (readCommitted && producerStateManager.firstUndecidedOffset().isPresent()
+                        && producerStateManager.firstUndecidedOffset().get().compareTo(offset) <= 0) {
+                    future.complete(ReadRecordsResult.error(Errors.NONE));
```

Comment:
Do we need to include aborted txns here?

Reply:
I think don't need to include aborted txns, the records is empty.